src/runner_test.go

```diff
@@ -7,85 +7,125 @@ import (
 )
 
 func TestProcessSignedScript(t *testing.T) {
-	shouldVerifyYaml := false
-	shouldDoInsightsCoreGPGCheck := false
-	temporaryWorkerDirectory := "test-dir"
-	config = &Config{
-		VerifyYAML:               &shouldVerifyYaml,
-		TemporaryWorkerDirectory: &temporaryWorkerDirectory,
-		InsightsCoreGPGCheck:     &shouldDoInsightsCoreGPGCheck,
-	}
-
-	defer os.RemoveAll(temporaryWorkerDirectory)
-
-	// Test case 1: verification disabled, no yaml data supplied = empty output
-	yamlData := []byte{}
-	expectedResult := ""
-	result := processSignedScript(yamlData)
-	if result != expectedResult {
-		t.Errorf("Expected %q, but got %q", expectedResult, result)
-	}
-
-	// Test case 2: verification disabled, yaml data supplied = non-empty output
-	yamlData = []byte(`
+	testCases := []struct {
+		name           string
+		verifyYAML     bool
+		yamlData       []byte
+		expectedResult string
+	}{
+		{
+			name:           "verification disabled, no yaml data supplied = empty output",
+			verifyYAML:     false,
+			yamlData:       []byte{},
+			expectedResult: "",
+		},
+		{
+			name:       "verification disabled, yaml data supplied = non-empty output",
+			verifyYAML: false,
+			yamlData: []byte(`
 vars:
-    _insights_signature: "invalid-signature"
-    _insights_signature_exclude: "/vars/insights_signature,/vars/content_vars"
+    insights_signature: "invalid-signature"
+    insights_signature_exclude: "/vars/insights_signature,/vars/content_vars"
     content: |
         #!/bin/sh
         echo "$RHC_WORKER_FOO $RHC_WORKER_BAR!"
     content_vars:
         FOO: Hello
-        BAR: World`)
-	expectedResult = "Hello World!\n"
-	result = processSignedScript(yamlData)
-	if result != expectedResult {
-		t.Errorf("Expected %q, but got %q", expectedResult, result)
+        BAR: World`),
+			expectedResult: "Hello World!\n",
+		},
+		{
+			name:       "verification enabled, invalid signature = error msg returned",
+			verifyYAML: true,
+			yamlData: []byte(`
+vars:
+    insights_signature: "invalid-signature"
+    insights_signature_exclude: "/vars/insights_signature,/vars/content_vars"
+    content: |
+        #!/bin/sh
+        echo "$RHC_WORKER_FOO $RHC_WORKER_BAR!"
+    content_vars:
+        FOO: Hello
+        BAR: World`),
+			expectedResult: "Signature of yaml file is invalid",
+		},
 	}
 
-	// FIXME: This is false success because verification fails on missing insighs-client
-	// Test case 3: verification enabled, invalid signature = error msg returned
-	shouldVerifyYaml = true
-	shouldDoInsightsCoreGPGCheck = true
-	expectedResult = "Signature of yaml file is invalid"
-	result = processSignedScript(yamlData)
-	if result != expectedResult {
-		t.Errorf("Expected %q, but got %q", expectedResult, result)
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			shouldVerifyYaml := tc.verifyYAML
+			shouldDoInsightsCoreGPGCheck := tc.verifyYAML // Assume the same value for simplicity
+			temporaryWorkerDirectory := "test-dir"
```

Comment:
Can we change this to use the `t.TempDir()`? Feels safer to rely on a temporary directory from the testing framework rather than a user defined directory.

Reply:
oops missed two, changed in in runner and also server test